tests/e2e/allowlists/allowlists_service_invocation_test.go

```diff
@@ -0,0 +1,169 @@
+// +build e2e
+
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package service_invocation_e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/dapr/dapr/tests/e2e/utils"
+	kube "github.com/dapr/dapr/tests/platforms/kubernetes"
+	"github.com/dapr/dapr/tests/runner"
+	"github.com/stretchr/testify/require"
+)
+
+type testCommandRequest struct {
+	RemoteApp        string `json:"remoteApp,omitempty"`
+	Method           string `json:"method,omitempty"`
+	RemoteAppTracing string `json:"remoteAppTracing"`
+}
+
+type appResponse struct {
+	Message string `json:"message,omitempty"`
+}
+
+const numHealthChecks = 60 // Number of times to call the endpoint to check for health.
+
+var tr *runner.TestRunner
+
+func TestMain(m *testing.M) {
+	// These apps will be deployed for hellodapr test before starting actual test
+	// and will be cleaned up after all tests are finished automatically
+	testApps := []kube.AppDescription{
+		{
+			AppName:        "allowlists-caller",
+			DaprEnabled:    true,
+			ImageName:      "e2e-service_invocation",
+			Replicas:       1,
+			IngressEnabled: true,
+		},
+		{
+			Config:         "allowlistsappconfig",
+			AppName:        "allowlists-callee-http",
+			DaprEnabled:    true,
+			ImageName:      "e2e-service_invocation",
+			Replicas:       1,
+			IngressEnabled: false,
+		},
+		{
+			Config:         "allowlistsgrpcappconfig",
+			AppName:        "allowlists-callee-grpc",
+			DaprEnabled:    true,
+			ImageName:      "e2e-service_invocation_grpc",
+			Replicas:       1,
+			IngressEnabled: false,
+			AppProtocol:    "grpc",
+		},
+	}
+
+	tr = runner.NewTestRunner("hellodapr", testApps, nil, nil)
+	os.Exit(tr.Start(m))
+}
+
+var serviceinvocationTests = []struct {
+	in               string
+	remoteApp        string
+	appMethod        string
+	expectedResponse string
+}{
+	{
+		"Test allow with callee side http",
+		"allowlists-callee-http",
+		"opAllow",
+		"opAllow is called",
+	},
+	{
+		"Test deny with callee side http",
+		"allowlists-callee-http",
+		"opDeny",
+		"rpc error: code = PermissionDenied desc = access control policy has denied access to appid: allowlists-caller operation: opDeny verb: POST",
+	},
+}
+
+var moreServiceinvocationTests = []struct {
```

Comment:
I would rename `moreServiceInvocationTests` to `allowlistTests`

Reply:
fixed